TrenDAP/Model/DataSet.cs

```diff
@@ -323,7 +329,30 @@ private Task<JObject> QuerySapphire(JObject jObject, DataSet dataset, DataSetJso
             }, cancellationToken);
 
         }
+        public static IEnumerable<HIDSPoint> ParsePoints(Task<HttpResponseMessage> hidsTask, CancellationToken cancellationToken)
+        {
+            HttpResponseMessage responseMessage = hidsTask.Result;
+            List<HIDSPoint> points = new List<HIDSPoint>();
+            using (Stream stream = responseMessage.Content.ReadAsStreamAsync().Result)
```

Comment:
When accessing the result of a task, it's typically recommended to use `task.GetAwaiter().GetResult()` instead of using `task.Result` directly. In the event that an exception is thrown, you should get a better stack trace that way.

As for this function, since it's self-contained and waiting on multiple async results, maybe it would be better to use async/await here move the synchronous wait up to the caller?

```c#
IEnumerable<HIDSPoint> points = ParsePointsAsync(rsp, cancellationToken)
    .GetAwaiter()
    .GetResult();
```

Better yet, every one of the callers seems like they could be adjusted to use async/await themselves. You shouldn't need to use these synchronous waits at all.

Reply:
The way we handle this changes in the next branch. To avoid loading everything on the webserver before sending up, I tried making a converter to an actionresult stream instead and sending that up. See here.

https://github.com/GridProtectionAlliance/TrenDAP/commit/ba4a4f4734b1be6b5ad600f28b55cd4c5b3c7cde